src/toniarts/openkeeper/world/WorldState.java

```diff
@@ -188,6 +191,26 @@ private void initPlayerCreatures() {
 
                 @Override
                 public void onSpawn(CreatureControl creature) {
+                    if (player.getId() == stateManager.getState(PlayerState.class).getPlayerId() && player.getCreatureControl().getTypeCount(creature.getCreature()) == 0) {

```

Comment:
Yes, that can be added to the PlayerCreatureControl since:
 - We actually should remove the entry when count == 0, so we can't use contains() to check this. Because of the UI ordering. The UI doesn't show 0 and I think it shows them in the order which the types spawned, minor detail

We should check how the original does the intros. My assumptions:
 - In campaign levels you get intro only for the totally new creature types -> if so, this could be as simple as checking the level type and if there is an override -> show intro, otherwise no
 - In skirmish/mp/mpd levels you always get the intro for the first time a type is introduced